evap/staff/tests/test_views.py

```diff
@@ -665,6 +665,31 @@ def test_view_downloads_excel_file(self):
         self.assertEqual(workbook.sheets()[0].row_values(0)[0],
                          'Evaluation {0}\n\n{1}'.format(self.semester.name, ", ".join([self.course_type.name])))
 
+    def test_heading_question_filtering(self):
+        course = mommy.make(Course, state='published', type=self.course_type, semester=self.semester)
+        contributor = mommy.make(UserProfile)
+        questionnaire = mommy.make(Questionnaire)
+
+        mommy.make(Question, type="H", questionnaire=questionnaire, order=0)
+        heading_question = mommy.make(Question, type="H", questionnaire=questionnaire, order=1)
+        likert_question = mommy.make(Question, type="L", questionnaire=questionnaire, order=2)
+        mommy.make(Question, type="H", questionnaire=questionnaire, order=3)
+
+        contribution = mommy.make(Contribution, course=course, questionnaires=[questionnaire], contributor=contributor)
+        mommy.make(RatingAnswerCounter, question=likert_question, contribution=contribution, answer=3, count=100)
+
+        page = self.app.get(self.url, user='staff')
+        form = page.forms["semester-export-form"]
+        form.set('form-0-selected_course_types', 'id_form-0-selected_course_types_0')
+        form.set('include_not_enough_answers', 'on')
+        response = form.submit()
+
+        workbook = xlrd.open_workbook(file_contents=response.content)
```

Comment:
can you check whether it's reasonably easy to test the exporter directly without using the view?

Reply:
Yes it's possible using this approach:
```
        binary_content = BytesIO()
        ExcelExporter(course.semester).export(binary_content, [[course.type.id]], True, True)
        binary_content.seek(0)
        workbook = xlrd.open_workbook(file_contents=binary_content.read())
```

I'd move the new test function to the export tests. Should I also move `test_view_excel_file_sorted` there? It also just tests whether the exporter correclty uses `course.name` to sort the courses. The name is localized by django. I don't quite know how to elegantly tell it which language to use if I don't go through the view.